backend/test_observer/controllers/test_executions/logic.py

```diff
@@ -49,3 +58,124 @@ def parse_c3_test_result_status(status: C3TestResultStatus) -> TestResultStatus:
             return TestResultStatus.FAILED
         case C3TestResultStatus.SKIP:
             return TestResultStatus.SKIPPED
+
+
+def get_historic_artefact_builds_query(
+    session: Session,
+    artefact: Artefact,
+    architecture: str,
+) -> RowReturningQuery:
+    """
+    Helper method to get a query that fetches the latest Artefact Build IDs
+    for given Artefact object identifiers (name, track, repo, store) and architecture.
+
+    The query only returns the latest revision build for each Artefact, in case there
+    are multiple revision of the same artefact version.
+
+    Parameters:
+        session (Session): Database session object used to generate the query for
+        artefact (Artefact): Artefact objects used to take the identifiers from
+        architecture (str): Architecture name to filter the results returned
+
+    Returns:
+        Prepared query object that can be executed as a query itself using .all()
+        or a subquery using .subquery()
+    """
+
+    return (
+        session.query(ArtefactBuild.id)
+        .distinct(ArtefactBuild.artefact_id)
```

Comment:
does this produce a `DISTINCT` or `DISTINCT ON` statement?

Reply:
Depending on the DB dialect, for PostgreSQL it produces `DISTINCT ON`, for other SQL databases it produces standard `DISTINCT`.

Here is the query it produces:
```
SELECT DISTINCT ON (artefact_build.artefact_id) artefact_build.id 
FROM artefact_build JOIN artefact ON artefact.id = artefact_build.artefact_id 
WHERE artefact.name = 'core22' AND artefact.track = '22' AND artefact.repo IS NULL AND artefact.series IS NULL AND artefact_build.architecture = 'armhf' ORDER BY artefact_build.artefact_id DESC, artefact_build.revision DESC
```